magento/magento2
# Elasticsearch price fieldname is incorrect during indexing when storeId and websiteId do not match

### Description (*)

The method getProductPriceData expects $websiteId as a parameter, but $storeId was given. It should be $websiteId since prices are website specific. Also see \Magento\Elasticsearch\Elasticsearch5\Model\Adapter\DataMapper\ProductDataMapper where a similar method exists and $websiteId is used. In cases where $websiteId and $storeId are different this causes the price filter to not work because ElasticSearch contains a field $fieldName = 'price_' . $customerGroupId . '_' . $storeId, but during filtering $fieldName = 'price_' . $customerGroupId . '_' . $websiteId is requested.


### Fixed Issues (if relevant)


### Manual testing scenarios (*)

1. Create a website with 2 stores (i.e. there is a store with storeId which differs from websiteId)
2. Enable elasticsearch as search backend
3. Enable index on schedule
4. run bin/magento indexer:reindex
5. Make sure the price attribute is enabled as a filter
6. The price filter should now work for both stores after this change

### Contribution checklist (*)
